libs/core/src/components/sage-textarea/test/sage-textarea.spec.tsx

```diff
@@ -0,0 +1,271 @@
+    });
+    expect(root).toEqualHtml(`
+      <sage-textarea placeholder-text="Placeholder...">
+        <mock:shadow-root>
+          <div class="sage-textarea">
+            <textarea class="sage-textarea__field" placeholder="Placeholder..."></textarea>
+          </div>
+        </mock:shadow-root>
+      </sage-textarea>
+    `);
+  });
+
+  it('renders readonly when property is passed', async () => {
+    const {root} = await newSpecPage({
+      components: [SageTextarea],
+      html: `<sage-textarea readonly="true"></sage-textarea>`,
+    });
+    expect(root).toEqualHtml(`
+      <sage-textarea readonly="true">
+        <mock:shadow-root>
+          <div class="sage-textarea">
+            <textarea class="sage-textarea__field" readonly=""></textarea>
+          </div>
+        </mock:shadow-root>
+      </sage-textarea>
+    `);
+  });
+
+  it('renders required when property is passed', async () => {
+    const {root} = await newSpecPage({
+      components: [SageTextarea],
+      html: `<sage-textarea required="true"></sage-textarea>`,
+    });
+    expect(root).toEqualHtml(`
+      <sage-textarea required="true">
+        <mock:shadow-root>
+          <div class="sage-textarea">
+            <textarea class="sage-textarea__field" required=""></textarea>
+          </div>
+        </mock:shadow-root>
+      </sage-textarea>
+    `);
+  });
+
+  it('renders textarea id and for attribute on label when property is passed', async () => {
+    const {root} = await newSpecPage({
+      components: [SageTextarea],
+      html: `<sage-textarea component-id="testId" label-text="label"></sage-textarea>`,
+    });
+    expect(root).toEqualHtml(`
+      <sage-textarea component-id="testId" label-text="label">
+        <mock:shadow-root>
+          <div class="sage-textarea">
+            <label htmlFor="testId">label</label>
+            <textarea class="sage-textarea__field" id="testId"></textarea>
+          </div>
+        </mock:shadow-root>
+      </sage-textarea>
+    `);
+  });
+
+
+  it('renders textarea value when property is passed', async () => {
+    const {root} = await newSpecPage({
+      components: [SageTextarea],
+      html: `<sage-textarea value="foo"></sage-textarea>`,
+    });
+    expect(root).toEqualHtml(`
+      <sage-textarea value="foo">
+        <mock:shadow-root>
+          <div class="sage-textarea">
+            <textarea class="sage-textarea__field">foo</textarea>
+          </div>
+        </mock:shadow-root>
+      </sage-textarea>
+    `);
+  });
+
+  it('updates value prop on value change', async () => {
+    const page = await newSpecPage({
+      components: [SageTextarea],
+      html: `<sage-textarea value="initial"></sage-textarea>`,
+    });
+
+    const sageTextarea = page.root;
+    expect(sageTextarea?.value).toBe('initial');
+
+    const textarea = sageTextarea?.shadowRoot?.querySelector<HTMLTextAreaElement>('textarea');
+    expect(textarea?.innerHTML.toString()).toBe('initial');
```

Comment:
Because the expected string value changes, does that make this an e2e test vs a unit test or am I incorrect?

Reply:
This is ok to do in a spec.